code/datums/chatmessage.dm

```diff
@@ -208,20 +214,20 @@
 	animate(message, alpha = 255, time = CHAT_MESSAGE_SPAWN_TIME)
 
 	// Register with the runechat SS to handle EOL and destruction
-	scheduled_destruction = world.time + (lifespan - CHAT_MESSAGE_EOL_FADE)
-	enter_subsystem()
+	var/duration = lifespan - CHAT_MESSAGE_EOL_FADE
+	fadertimer = addtimer(CALLBACK(src, .proc/end_of_life), duration, TIMER_STOPPABLE|TIMER_DELETE_ME, SSrunechat)
 
 /**
  * Applies final animations to overlay CHAT_MESSAGE_EOL_FADE deciseconds prior to message deletion,
- * sets time for scheduling deletion and re-enters the runechat SS for qdeling
+ * sets timer for scheduling deletion
  *
  * Arguments:
  * * fadetime - The amount of time to animate the message's fadeout for
  */
 /datum/chatmessage/proc/end_of_life(fadetime = CHAT_MESSAGE_EOL_FADE)
-	eol_complete = scheduled_destruction + fadetime
+	isFading = TRUE
 	animate(message, alpha = 0, time = fadetime, flags = ANIMATION_PARALLEL)
-	enter_subsystem(eol_complete) // re-enter the runechat SS with the EOL completion time to QDEL self
+	addtimer(CALLBACK(src, .proc/Destroy), fadetime, TIMER_DELETE_ME, SSrunechat)
```

Comment:
Should we use QDEL_IN here? Or this solution is ok?

Reply:
Not using QDEL_IN is fine if you want it to be on SSrunechat (though we could make a new macro for that), butdon't call `Destroy` directly at the very least. Use `GLOBAL_PROC, /proc/qdel, src` instead.

Reply:
https://github.com/tgstation/tgstation/pull/64366/commits/bd6a9a1645c88b874434e78eec37e246e3673f1c